frontend/src/views/BookerView/components/SlotSelectionUserInfo.vue

```diff
@@ -0,0 +1,155 @@
+<script setup lang="ts">
+import { computed, ref, useTemplateRef } from 'vue';
+import { useI18n } from 'vue-i18n';
+import { storeToRefs } from 'pinia';
+import { PrimaryButton, TextInput } from '@thunderbirdops/services-ui';
+import { useUserStore } from '@/stores/user-store';
+import { useBookingViewStore } from '@/stores/booking-view-store';
+import { HTMLInputElementEvent } from '@/models';
+import { AlertSchemes } from '@/definitions';
+import AlertBox from '@/elements/AlertBox.vue';
+
+enum Tab {
+  GUEST,
+  SIGN_IN
+}
+
+const { t } = useI18n();
+const userStore = useUserStore();
+const bookingViewStore = useBookingViewStore();
+const userLoginFormRef = useTemplateRef('user-login-form');
+
+const { guestUserInfo, guestUserInfoValid } = storeToRefs(bookingViewStore)
+
+const selectedTab = ref<Tab>(Tab.GUEST);
+const userLoginEmail = ref('');
+const userLoginPassword = ref('');
+const userLoginError = ref('');
+const userLoginLoading = ref(false);
+
+const guestUserName = computed({
+  get: () => guestUserInfo.value.name,
+  set: (value) => {
+    bookingViewStore.$patch({ guestUserInfo: { name: value } })
+  }
+})
+
+const guestUserEmail = computed({
+  get: () => guestUserInfo.value.email,
+  set: (value) => {
+    bookingViewStore.$patch({ guestUserInfo: { email: value } })
+  }
+})
+
+const onGuestUserEmailChanged = (e: HTMLInputElementEvent) => {
+  guestUserInfoValid.value = e.target.checkValidity();
+}
+
+const onLogin = async () => {
+  userLoginError.value = '';
+  userLoginLoading.value = true;
+
+  if (userLoginFormRef.value.checkValidity()) {
+    const { error } = await userStore.login(userLoginEmail.value, userLoginPassword.value);
+
+    if (error) {
+      // FIXME: The return of the login function is not standardized at the moment
+      // and currently inferred as string | boolean but it does return an object
+      // with a detail key in this case
```

Comment:
Good catch, I also had troubles with this in the past. Maybe we can create a ticket for improving the error types / error handling.